aws private endpoint deployment (#1024)
arangodb/docs: oasis/deployments.md

   **Connections** will increase and you can view the connection details by
   clicking it.

### Amazon Web Services (AWS)

AWS offers a feature called [AWS PrivateLink](https://aws.amazon.com/privatelink){:target="_blank"}
that enables you to privately connect your VPC to services, without exposure to
the internet. You can control the specific API endpoints, sites, and services
that are reachable from your VPC.

Amazon Virtual Private Cloud (VPC) allows you to launch AWS resources into a
virtual network that you have defined. It closely resembles a traditional
network that you would normally operate, with the benefits of using the AWS
scalable infrastructure. 

You can
[create a regular deployment](#how-to-create-a-new-deployment) and change it
to a private endpoint deployment afterwards.

> The deployment will not be reachable from the internet anymore, other than via

Reviewer: ```suggestion
The deployment will not be exposed to public internet anymore, other than via
```
Would this be more relatable?

the Oasis dashboard to administrate it. To revert it to a public deployment,
please contact the support team via __Request help__ in the help menu.

To configure a Private Endpoint, you need to provide the AWS principals related
to your VPC. The ArangoDB Oasis platform configures a Private Endpoint Service
that automatically connects to Private Endpoints that are created in those principals. 

1. Open the deployment you want to change.
2. In the **Overview** tab, click the **Edit** button with an ellipsis (`…`)
   icon. If you see a pencil icon and no menu opens, then you are either on the
   free-to-try or professional tier, or the selected deployment is not eligible.
   {% hint 'info' %}
   The private endpoint option is only available on the enterprise tier for
   deployments running on AWS.
   {% endhint %}
3. Click **Change to private endpoint** in the menu.

>    ![Oasis Deployment AWS Change to Private Endpoint](images/oasis-aws-change-to-private-endpoint.png)

Author: Already included.

4. In the configuration wizard, click **Next** to enter your configuration details.
5. Click **Add Principal** to start configuring the AWS principal(s). 
   You need to enter a valid account, which is your 12 digit AWS account ID.
   Adding usernames or role names is optional. You can also
   skip this step and add them later from the summary view.
   {% hint 'info' %}
   Principals cannot be changed anymore once a connection has been established.
   {% endhint %}
   ![Oasis AWS Private Endpoint Configure Principals](images/oasis-aws-endpoint-configure-principals.png)

> 6. You may enter one or more Alternate DNS names. This step is optional, you can 

Author: Included the screenshot.

   add or change them later. Click **Next** to continue.
7. Confirm that you want to use a Private Endpoint for your deployment by
   clicking **Confirm Settings**.
8. Back in the **Overview** tab, scroll down to the **Private Endpoint** section
   that is now displayed to see the connection status and change the
   configuration, if needed.
   ![Oasis AWS Private Endpoint Overview](images/oasis-aws-private-endpoint-overview.png)

> 

Reviewer 2: This describes the steps to take on Oasis, should we include the next steps a customer has to do the establish a connection as well or point him to the AWS documentation?

See: https://docs.aws.amazon.com/vpc/latest/privatelink/create-endpoint-service.html#share-endpoint-service
paragraph: 'Connect to an endpoint service as the service consumer'

   **Connections** will increase and you can view the connection details by
   clicking it.

### Amazon Web Services (AWS)

AWS offers a feature called [AWS PrivateLink](https://aws.amazon.com/privatelink){:target="_blank"}

> that enables you to privately connect your VPC to services, without exposure to

Reviewer 3: please define the acronym the first time you use it.

Author: done

5. Click **Add Principal** to start configuring the AWS principal(s). 
   You need to enter a valid account, which is your 12 digit AWS account ID.
   Adding usernames or role names is optional. You can also
   skip this step and add them later from the summary view.
   {% hint 'info' %}
   Principals cannot be changed anymore once a connection has been established.
   {% endhint %}
   ![Oasis AWS Private Endpoint Configure Principals](images/oasis-aws-endpoint-configure-principals.png)
6. You may enter one or more Alternate DNS names. This step is optional, you can 
   add or change them later. Click **Next** to continue.
   ![Oasis AWS Private Endpoint Alternate DNS](images/oasis-aws-private-endpoint-dns.png)
7. Confirm that you want to use a Private Endpoint for your deployment by
   clicking **Confirm Settings**.
8. Back in the **Overview** tab, scroll down to the **Private Endpoint** section
   that is now displayed to see the connection status and change the
   configuration, if needed.
   ![Oasis AWS Private Endpoint Overview](images/oasis-aws-private-endpoint-overview.png)
   {% hint 'tip' %}

>    To understand more or request help from the Oasis support team, click **Help**

Reviewer 3: To learn more or request help...

Author: done

>    at the top right corner of the **Private Endpoint** section.

Reviewer 3: in the top right corner

Author: done
